src/modules/battery.cpp

```diff
@@ -116,21 +117,17 @@ auto waybar::modules::Battery::update() -> void
 {
   auto [capacity, status] = getInfos();
   label_.set_tooltip_text(status);
-  bool charging = status == "Charging";
+  std::transform(status.begin(), status.end(), status.begin(), ::tolower);
   auto format = format_;
-  if (charging) {
-    label_.get_style_context()->add_class("charging");
-    if (config_["format-charging"].isString()) {
-      format = config_["format-charging"].asString();
-    }
-  } else {
-    label_.get_style_context()->remove_class("charging");
-    if (status == "Full" && config_["format-full"].isString()) {
-      format = config_["format-full"].asString();
-    }
-  }
-  auto state = getState(capacity, charging);
-  if (!state.empty() && config_["format-" + state].isString()) {
+  auto state = getState(capacity);
+  label_.get_style_context()->remove_class(old_status_);
+  label_.get_style_context()->add_class(status);
+  old_status_ = status;
+  if (!state.empty() && config_["format-" + status + "-" + state].isString()) {
+    format = config_["format-" + status + "-" + state].asString();
+  }else if (config_["format-" + status].isString()) {
+    format = config_["format-" + status].asString();
+  }else if (!state.empty() && config_["format-" + state].isString()) {
```

Comment:
Space

Reply:
It's hard to break old habits. Fixed. (+Initialization for status in getInfos(), so we don't get an empty status on Unknown).